### PR TITLE
fix #2589

### LINKED ISSUE #2589: Error pulling docker image (tag regex fails)
I'm trying to use `quay.io/biocontainers/platypus-variant:0.8.1.1--htslib1.5_0`, but I'm getting the following error:

```
[ERROR] [08/31/2017 20:01:21.193] [cromwell-system-akka.dispatchers.engine-dispatcher-7] [akka://cromwell-system/user/SingleWorkflowRunnerActor/WorkflowManage
rActor] WorkflowManagerActor Workflow 73494fe9-ef7b-45e0-b982-b50baf3281d7 failed (during ExecutingWorkflowState): Docker image quay.io/biocontainers/platypus-variant:0.8.1.1--htslib1.5_0 has an invalid syntax.
java.lang.IllegalArgumentException: Docker image quay.io/biocontainers/platypus-variant:0.8.1.1--htslib1.5_0 has an invalid syntax.
        at cromwell.docker.DockerImageIdentifier$.fromString(DockerImageIdentifier.scala:60)
        at cromwell.engine.workflow.lifecycle.execution.preparation.JobPreparationActor.handleDockerValue$1(JobPreparationActor.scala:112)
        at cromwell.engine.workflow.lifecycle.execution.preparation.JobPreparationActor.cromwell$engine$workflow$lifecycle$execution$preparation$JobPreparatio
nActor$$fetchDockerHashesIfNecessary(JobPreparationActor.scala:129)
        at cromwell.engine.workflow.lifecycle.execution.preparation.JobPreparationActor$$anonfun$1.applyOrElse(JobPreparationActor.scala:57)
        at cromwell.engine.workflow.lifecycle.execution.preparation.JobPreparationActor$$anonfun$1.applyOrElse(JobPreparationActor.scala:54)
        at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:34)
        at akka.actor.FSM.processEvent(FSM.scala:663)
        at akka.actor.FSM.processEvent$(FSM.scala:660)
        at cromwell.engine.workflow.lifecycle.execution.preparation.JobPreparationActor.processEvent(JobPreparationActor.scala:33)
        at akka.actor.FSM.akka$actor$FSM$$processMsg(FSM.scala:657)
        at akka.actor.FSM$$anonfun$receive$1.applyOrElse(FSM.scala:651)
        at akka.actor.Actor.aroundReceive(Actor.scala:513)
        at akka.actor.Actor.aroundReceive$(Actor.scala:511)
        at cromwell.engine.workflow.lifecycle.execution.preparation.JobPreparationActor.aroundReceive(JobPreparationActor.scala:33)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:527)
        at akka.actor.ActorCell.invoke(ActorCell.scala:496)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
        at akka.dispatch.Mailbox.run(Mailbox.scala:224)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
        at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

```

### PR DESCRIPTION
**Pull request by raylim**

### CODE REVIEW

File: dockerHashing/src/main/scala/cromwell/docker/DockerImageIdentifier.scala [hunk 1]
```diff
@@ -42,7 +42,7 @@ object DockerImageIdentifier {
           :                                     # Tag separator. ':' is followed by a tag
        
           (                                     # Begin capturing group for reference 
-            [A-Za-z0-9]+(?:[.:_-][A-Za-z0-9]+)* # Reference
+            [A-Za-z0-9]+(?:[.:_\-A-Za-z0-9]+)*  # Reference
```
Comment (mcovarr) [inline dockerHashing/src/main/scala/cromwell/docker/DockerImageIdentifier.scala]: if you put the dash first in the character class you shouldn't have to escape it:

```
[A-Za-z0-9]+(?:[.:_\-A-Za-z0-9]+)* // original
[A-Za-z0-9]+(?:[-.:_A-Za-z0-9]+)* // proposed
```

Comment (mcovarr) [reply dockerHashing/src/main/scala/cromwell/docker/DockerImageIdentifier.scala]: Also we should probably add a case to the existing list in `DockerImageIdentifierSpec` that confirms this works as expected.

**Comment by mcovarr** [comment]
Thanks very much for the error report and fix!  😄  :+1:

**Comment by Horneth** [comment]
Yes I think the `-` needs to be escaped.
If you could also make the same change on line 52, and add  a new line with your docker image in the table in `DockerImageIdentifierSpec` to validate that it works it would be great :)

Thanks !

**Comment by Horneth** [comment]
Awesome ! 👍

**Pull request merged**